packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -56,10 +73,85 @@ class UnthemedGraphNG extends React.Component<GraphNGProps, GraphNGState> {
     this.state = {
       alignedDataFrame: alignedData,
       data: preparePlotData(alignedData),
-      config: preparePlotConfigBuilder(alignedData, props.theme, this.getTimeRange, this.getTimeZone),
     };
   }
 
+  componentDidMount() {
+    const { alignedDataFrame } = this.state;
+
+    this.panelContext = this.context as PanelContext;
+    const { eventBus } = this.panelContext;
+
+    const config = preparePlotConfigBuilder(
+      alignedDataFrame,
+      this.props.theme,
+      this.getTimeRange,
+      this.getTimeZone,
+      eventBus
+    );
+    this.setState({ config });
+
+    this.subscriptions.push(
+      eventBus
+        .getStream(LegacyGraphHoverEvent)
+        .pipe(
+          throttleTime(50),
+          filter((e) => e.origin !== eventBus)
+        )
+        .subscribe({
+          next: (evt) => {
+            const u = this.plotInstance?.current;
+            if (u) {
+              // Try finding left position on time axis
+              const left = u.valToPos(evt.payload.point.time, 'time');
+              console.log(left);
+              let top;
+              if (left) {
+                // find midpoint between points at current idx
+                top = findMidPointYPosition(u, u.posToIdx(left));
+              }
+
+              if (!top || !left) {
+                return;
+              }
+
+              u.setCursor({
+                left,
+                top,
+              });
+            }
+          },
+        })
+    );
+
+    this.subscriptions.push(
+      eventBus
+        .getStream(LegacyGraphHoverClearEvent)
+        .pipe(
+          throttleTime(50),
+          filter((e) => e.origin !== eventBus)
+        )
+        .subscribe({
+          next: (evt) => {
+            const u = this.plotInstance?.current;
+
+            if (u) {
+              u.setCursor({
+                left: -10,
+                top: -10,
+              });
+            }
+          },
+        })
+    );
+  }
```

Comment:
I think you can define `this.subscriptions` as `Subscription` instead and then you should new it up in the constructor like 
```
this.subscriptions = new Subscription();
```
After that you can just call `this.subscriptions.add(...)` here and when you want to unsubscribe them you can just call `this.subscriptions.unsubscribe()` and all subscriptions will be unsubscribed.